taipy/core/job/job.py

```diff
@@ -148,36 +148,107 @@ def creation_date(self, val):
-        self._execution_ended_at = val
+        Returns:
+            Optional[datetime]: The date time when the job was run.
+                If the job is not run, None is returned.
+        """
+        return self._status_change_records.get("RUNNING", None)
+
+    @property
+    @_self_reload(_MANAGER_NAME)
+    def finished_time(self) -> Optional[datetime]:
+        """Get the date time when the job was finished.
+
+        Returns:
+            Optional[datetime]: The date time when the job was finished.
+                If the job is not finished, None is returned.
+        """
+        if self.is_finished():
+            if self.is_completed():
+                return self._status_change_records["COMPLETED"]
+            elif self.is_failed():
+                return self._status_change_records["FAILED"]
+            elif self.is_canceled():
+                return self._status_change_records["CANCELED"]
+            elif self.is_skipped():
+                return self._status_change_records["SKIPPED"]
+            elif self.is_abandoned():
+                return self._status_change_records["ABANDONED"]
+
+        return None
 
     @property
     @_self_reload(_MANAGER_NAME)
     def execution_duration(self) -> Optional[float]:
         """Get the duration of the job execution in seconds.
+        The execution time is the duration from the job running to the job completion.
 
         Returns:
-            Optional[float]: The duration of the job execution in seconds. If the job is not
-            completed, None is returned.
+            Optional[float]: The duration of the job execution in seconds.
+                - If the job was not run, None is returned.
+                - If the job is not finished, the execution time is the duration
+                  from the running time to the current time.
         """
-        if self._execution_started_at and self._execution_ended_at:
-            return (self._execution_ended_at - self._execution_started_at).total_seconds()
-        return None
+        if "RUNNING" not in self._status_change_records:
+            return None
+
+        if self.is_finished():
+            return (self.finished_time - self._status_change_records["RUNNING"]).total_seconds()
+
+        return (datetime.now() - self._status_change_records["RUNNING"]).total_seconds()
+
+    @property
+    @_self_reload(_MANAGER_NAME)
+    def pending_duration(self) -> Optional[float]:
+        """Get the duration of the job in the pending state in seconds.
+
+        Returns:
+            Optional[float]: The duration of the job in the pending state in seconds.
+                - If the job is not running, None is returned.
+                - If the job is not pending, the pending time is the duration
+                  from the submission to the current time.
+        """
+        if "PENDING" not in self._status_change_records:
+            return None
+
+        if self.is_finished() or self.is_running():
```

Comment:
hmmm this might cause alot of reload yeah?

Reply:
it's an attribute so I think it's reasonable